analyze_model.py

```diff
@@ -54,3 +61,24 @@ def get_weight_distances(model, source_weights_model, layer_indices=[], norm_ord
                     distance_list[layer].append(
                         np.linalg.norm(model_flatten_weights - source_flatten_weights, ord=order))
         return distance_list
+
+    @staticmethod
+    def generate_heatmap(data, row_labels, col_labels, filename, output_dir, verbose=True):
```

Comment:
I don't like how this function needs an `output_dir`, the point of `ResourceManager` is that it should be the only class that knows that paths etc... Maybe keep the `generate_heatmap` part in `Experiment` and maybe put the logic in `AnalyzeModel`:

```
class Experiment:
    ...
    def generate_heatmap(self, data, rows, cols):
        figure = AnalyzeModel.generate_heatmap_figure(data, rows, cols)
        figure.savefig(self._results_dir + filename)